### ISSUE TITLE
Documentation: Development Quick-start via Docker

### ISSUE THREAD

**Issue by oxr463** [buildbot/buildbot#5100]
> In order to develop Buildbot you need just a python environment and possibly some native packages in stripped-down setups. The most up to date list in the docker file we use to manage our CI (MetaBBotDockerFile).

How about providing instructions on developing directly from Docker?

See also: https://github.com/buildbot/buildbot/issues/3916

**Comment by tardyp** [comment]
I don't think it is really practical to develop with docker, you have to configure file system, the editor do not have access to the virtualenv, so you cannot use their python langage server or similar nifties.

Our goal is to attract developers in longer term, show that it is not that difficult to work on buildbot. 
So showing a way to hack on Buildbot which only allows very small hacks might be counter productive.

**Comment by oxr463** [comment]
I disagree, since I pretty much exclusively develop within a Docker container.

Regarding the file system and editing from an IDE,

```sh
git clone https://github.com/buildbot/buildbot.git
docker run -v $(pwd)/buildbot:/tmp/buildbot -it buildbot/dev
```

The only dependencies required for this would be Docker itself.

**Comment by tardyp** [comment]
Yes, but in that case, the IDE do cannot access the virtualenv ( or rather the virtualenv is pointing to shared libs from the docker root, and thus is unusable in the host environment)
So the IDE won't be able to show pylint issues on the flight, or do autocompletion, etc.

I admit that some people do not find these useful, and docker dev env will work for them.
If you want to try yourself at it, we can work on that.

The "official" docker image for building buildbot, is the one we use on our CI:
https://hub.docker.com/r/buildbot/metabbotcfg
Its a bit large, but it is garanteed to always work.

The build process is then "documented" in the CI config in .bbtravis.yml

**Comment by oxr463** [comment]
Interesting, which IDE requires this? Perhaps I can test it out, (I use Vim).

**Comment by tardyp** [comment]
Visual Studio Code, PyCharm are the one we use @work, besides vim..
